packages/nimble-components/src/table/index.ts

```diff
@@ -142,6 +157,20 @@ export class Table<
         this.refreshRows();
     }
 
+    // This is a workaround for dealing with a change in Safari 16+. It seems related to this bug, filed
+    // against WebKit: https://bugs.webkit.org/show_bug.cgi?id=246899, which was closed as "intentional"
+    // behavior change. The issue we faced was that using a filter for the `children` directive in the
+    // template that required JS object APIs (like instanceOf) did not work because the elements being
+    // filtered were not fully initialized JS objects.
+    // Note that the FAST DataGrid does something similar to this (using DOM.queueUpdate) to update its row indexes.
+    private updateColumnsFromChildren(): void {
+        DOM.queueUpdate(() => {
```

Comment:
Sorry I'm having a hard time tracing the discussion from the webkit issue of documentTarget through to essentially performing a requestAnimationFrame via queueUpdate.

I read that as just an arbitrary 16ms wait, I'm not sure what guarantees that gives us. If you can help me understand why we feel confident that will put us in a good state we can try it.

If we want a strong assertion to check for explicitly like all of the `childItems` are defined custom element and then we assign them to columns then I think we need something like the following: https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/whenDefined#examples

I think the workflow is something like: childItemsChanges -> await all custom element childItems upgrades -> update table columns with a filtered list of childItems

Reply:
Updated to use the APIs you point out.